sim/data_provider.py

```diff
@@ -0,0 +1,37 @@
+# pylint: disable=too-many-arguments, too-many-locals
+
+from abc import ABC, abstractmethod, abstractproperty
+from typing import Union
+import xarray as xr
+
+
+class IDataProvider(ABC):
+    """Interface for all data providers.
+    A DataProvider serves as the backing data for simulation
+    """
+
+    @abstractproperty
+    def settable(self) -> bool:
+        """Indicates whether this data provider allows its value to
+        be set by calling set_value
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def set_value(self, value: float) -> None:
```

Comment:
It would probably be more pytonic to use a setter for the property as in

```
@value.setter
@abstractmethod
def value(self, value: float) -> None:
    raise NotImplementedError
```

And move it after the 

```